open-build-service-api/src/lib.rs

```diff
@@ -170,6 +171,39 @@ pub struct Directory {
     pub linkinfo: Vec<LinkInfo>,
 }
 
+#[derive(Deserialize, Serialize, Debug)]
+pub struct CommitEntry {
+    pub name: String,
+    pub md5: String,
+}
```

Comment:
any reason why all fields are public here: rather then being opaque ?

Reply:
no particular reason; it's a pretty light / low-state struct so I didn't feel like being opaque was necessary, but I can add it if desired

Reply:
(also since we're using it in two places, one being a "setter" context)

Reply:
After going over it a bit again, I do feel like, because this is also in an API response, having the members public matches with other API responses and is a bit clearer to use. That being said, I added some builder-esque methods to CommitFileList that allow you to add files without actually creating a CommitEntry, which might help clean things up a bit.